### PR TITLE
extract executesc operation info and-send it to wails popup

### PR DESCRIPTION
**Pull request by B-Naoufal**

### CODE REVIEW

File: internal/handler/wallet/sign.go [hunk 1]
```diff
@@ -71,19 +74,18 @@ func (s *walletSign) Handle(params operations.SignParams) middleware.Responder {
...
 	if params.Body.CorrelationID != nil {
 		correlationId, resp = handleWithCorrelationId(wlt, params, s.gc)
 	} else {
 
 		var promptRequest prompt.PromptRequest
-		if callSC != nil {
+		// Check if the decoded message is a CallSC
+		if callSC, err := callsc.DecodeMessage(decodedMsg); err == nil && callSC != nil {
+			// Handle the case of a CallSC message
+			fmt.Println("Decoded CallSC message successfully")
```
Comment (Thykof) [inline internal/handler/wallet/sign.go:87]:
```suggestion
```

File: internal/handler/wallet/sign.go [hunk 2]
```diff
@@ -96,7 +98,23 @@ func (s *walletSign) Handle(params operations.SignParams) middleware.Responder {
 					Function:      callSC.Function,
 				},
 			}
+		} else if executeSC, err := executesc.DecodeMessage(decodedMsg); err == nil && executeSC != nil {
+			// Handle the case of an ExecuteSC message
+			fmt.Println("Decoded ExecuteSC message successfully")
```
Comment (Thykof) [inline internal/handler/wallet/sign.go:103]:
```suggestion
```

File: internal/handler/wallet/sign.go
Comment (Thykof) [inline internal/handler/wallet/sign.go:1]: the function `func (s *walletSign) Handle(params operations.SignParams) middleware.Responder` is too long and too complex. Please split it into multiple small functions to improve code readability and maintainability.

**Review by Thykof — CHANGES_REQUESTED** [review]

File: internal/handler/wallet/sign.go [hunk 3]
```diff
@@ -143,6 +115,67 @@ func (s *walletSign) Handle(params operations.SignParams) middleware.Responder {
...
+		promptRequest = s.prepareCallSCPromptRequest(callSC, wlt)
+	} else {
+		executeSC, err := executesc.DecodeMessage(decodedMsg)
+		if err == nil && executeSC != nil {
+			promptRequest = s.prepareExecuteSCPromptRequest(executeSC, wlt)
+		} else {
+			promptRequest = s.prepareUnknownPromptRequest(wlt)
+		}
+	}
+	return promptRequest, nil
```
Comment (Thykof) [inline internal/handler/wallet/sign.go:139]:
```suggestion
	callSC, err := callsc.DecodeMessage(decodedMsg)
	if err == nil && callSC != nil {
		promptRequest = s.prepareCallSCPromptRequest(callSC, wlt)
		return promptRequest, nil
	}
	executeSC, err := executesc.DecodeMessage(decodedMsg)
	if err == nil && executeSC != nil {
		promptRequest = s.prepareExecuteSCPromptRequest(executeSC, wlt)
		return promptRequest, nil
	}

	promptRequest = s.prepareUnknownPromptRequest(wlt)

	return promptRequest, nil
```

**Review by Thykof — CHANGES_REQUESTED** [review]

**Review by Thykof — APPROVED** [review]

**Review by gregLibert — APPROVED** [review]

**Pull request merged**